src/serial-console/azext_serialconsole/custom.py

```diff
@@ -0,0 +1,723 @@
+            func = self.send_reset
+            success_message = "Successfully Hard Reset VM      \r\n"
+            failure_message = "Failed to Hard Reset VM         \r\n"
+            loading_text = "Forcing VM to Hard Reset"
+        elif command == "sysrq" and arg_characters is not None:
+            def wrapper():
+                return self.send_sys_rq(arg_characters)
+            func = wrapper
+            success_message = "Successfully sent SysRq command\r\n"
+            failure_message = "Failed to send SysRq command. Make sure the input only contains numbers and letters.\r\n"
+            loading_text = "Sending SysRq to VM"
+        else:
+            return
+
+        GV.loading = True
+
+        th1 = threading.Thread(
+            target=self.send_loading_message, args=(loading_text,))
+        th1.daemon = True
+        th1.start()
+
+        if self.load_websocket_url():
+            def on_message(ws, _):
+                GV.trycount += 1
+                if func():
+                    GV.loading = False
+                    GV.terminating_app = True
+                    print(success_message, end="")
+                    ws.close()
+                elif GV.trycount >= 2:
+                    GV.loading = False
+                    GV.terminating_app = True
+                    print(failure_message, end="")
+                    ws.close()
+
+            wsapp = websocket.WebSocketApp(
+                self.websocket_url + "?authorization=" + self.access_token, on_message=on_message)
+            wsapp.run_forever()
+            GV.loading = False
+            if GV.trycount == 0:
+                error_message = "Could not establish connection to VM or VMSS."
+                recommendation = 'Try restarting it with "az vm restart".'
+                raise AzureConnectionError(
+                    error_message, recommendation=recommendation)
+        else:
+            GV.loading = False
+            error_message = "Could not establish connection to VM or VMSS."
+            recommendation = "Make sure the parameters name/resource-group/vmss-instance are correct."
+            raise ResourceNotFoundError(
+                error_message, recommendation=recommendation)
+
+
+def check_serial_console_enabled(cli_ctx):
+    subscription_id = get_subscription_id(cli_ctx)
+    connection_url = (f"{ARM_ENDPOINT}/subscriptions/{subscription_id}"
+                      f"/providers/{RP_PROVIDER}/consoleServices/"
+                      f"/default?api-version=2018-05-01")
+    token_info, _, _ = Profile().get_raw_token()
+    access_token = token_info[1]
+    application_json_format = "application/json"
+    headers = {'authorization': "Bearer " + access_token,
+               'accept': application_json_format,
+               'content-type': application_json_format}
+    result = requests.get(connection_url, headers=headers)
+    json_results = json.loads(result.text)
+    if result.status_code == 404:
```

Comment:
You should create a client to handle http requests. Azure core already implemented some usefully policies. Please following the example client. [https://github.com/Azure/azure-cli-extensions/blob/c353c4e1b41c47ff8fee68dc34b4c369de7738da/src/webpubsub/azext_webpubsub/vendored_sdks/azure_messaging_webpubsubservice/__init__.py#L131-L165](https://github.com/Azure/azure-cli-extensions/blob/c353c4e1b41c47ff8fee68dc34b4c369de7738da/src/webpubsub/azext_webpubsub/vendored_sdks/azure_messaging_webpubsubservice/__init__.py#L131-L165)